eng/pipelines/templates/stages/archetype-rust-release.yml

```diff
@@ -84,33 +85,77 @@ stages:
-                  env:
-                    CARGO_REGISTRY_TOKEN: $(azure-sdk-cratesio-token)
+                  - pwsh: |
+                      Write-Host "##vso[task.setvariable variable=ArtifactIndex]0"
+                    displayName: Set ArtifactIndex to 0
+
+                  - ${{ each artifact in parameters.Artifacts }}:
+                    - pwsh: |
+                        # Read artifact release order from release-order.json 
+                        # and use ArtifactIndex to select the right one
+                        $index = [int]'$(ArtifactIndex)'
+                        $artifacts = Get-Content '$(Pipeline.Workspace)/drop/release-order.json' | ConvertFrom-Json
+                        if ($index -ge $artifacts.Count) {
+                          Write-Error "ArtifactIndex $index is out of range (0..$($artifacts.Count - 1))"
+                          exit 1
+                        }
+
+                        $artifactName = $artifacts[$index]
+                        Write-Host "Releasing artifact $artifactName (index $index)"
+
+                        $artifactRootPath = '$(Pipeline.Workspace)/drop'
+                        $outDir = '$(Pipeline.Workspace)/esrp-release'
+
+                        if (Test-Path $outDir) {
+                          Write-Host "Cleaning output directory: $outDir"
+                          Remove-Item -Path $outDir -Recurse -Force
+                        }
+                        New-Item -ItemType Directory -Path $outDir -Force | Out-Null
+
+                        Write-Host "Artifact name: $artifactName"
+
+                        $packageMetadataPath = "$artifactRootPath/PackageInfo/$artifactName.json"
+                        if (!(Test-Path $packageMetadataPath)) {
+                          Write-Error "Package metadata file not found: $packageMetadataPath"
+                          exit 1
+                        }
+
+                        $packageMetadata = Get-Content -Raw $packageMetadataPath | ConvertFrom-Json
+                        $packageVersion = $packageMetadata.version
+                        Write-Host "Package version: $packageVersion"
+
+                        $cratePath = "$artifactRootPath/$artifactName/$artifactName-$packageVersion.crate"
+                        Copy-Item `
+                          -Path $cratePath `
+                          -Destination $outDir
+                        Write-Host "Contents of $outDir"
+                        Get-ChildItem -Path $outDir | ForEach-Object { Write-Host $_.FullName }
+                      displayName: 'Copy crate for ESRP'
+
+                    - task: EsrpRelease@10
+                      displayName: 'ESRP Release'
+                      inputs:
+                        connectedservicename: 'Azure SDK PME Managed Identity'
+                        ClientId: '5f81938c-2544-4f1f-9251-dd9de5b8a81b'
+                        DomainTenantId: '975f013f-7f24-47e8-a7d3-abc4752bf346'
```

Comment:
Should these be hardcoded? Not an SFI concern, but seems like they should be in a variable group or something for easy updates if necessary.

Reply:
These have been hardcoded historically